util/include/Serializable.h

```diff
@@ -37,72 +40,199 @@ typedef std::unique_ptr<char[], std::default_delete<char[]>> UniquePtrToChar;
 typedef std::unique_ptr<unsigned char[], std::default_delete<unsigned char[]>> UniquePtrToUChar;
 typedef std::shared_ptr<Serializable> SerializablePtr;
 
-//TODO [TK] remove
-class MemoryBasedBuf : public std::basic_streambuf<char> {
- public:
-  MemoryBasedBuf(const UniquePtrToChar &buf, size_t size) {
-    setg(buf.get(), buf.get(), buf.get() + size);
-  }
-};
-
-// This class allows usage of a regular buffer as a stream.
-// TODO [TK] remove
-class MemoryBasedStream : public std::istream {
- public:
-  MemoryBasedStream(const UniquePtrToChar &buf, size_t size) : std::istream(&buffer_), buffer_(buf, size) {
-    rdbuf(&buffer_);
-  }
-
- private:
-  MemoryBasedBuf buffer_;
-};
-
 class Serializable {
- public:
-
-  Serializable();
+public:
+  Serializable() = default;
   virtual ~Serializable() = default;
+  /**
+   * serialization API
+   */
+  template<typename T>
+  static void serialize(std::ostream& outStream, const T& t) { serialize_impl(outStream, t, int{}); }
```

Comment:
Pardon my ignorance on template type defs. What is the `int{}` for here?

Reply:
Here int is a third argument with no meaning, just for a signature

Reply:
Thanks!